### PR TITLE
app.js: Add url search for project

### PR DESCRIPTION
**Pull request by hemangsk**
Add search parameter whose value
goes in the search box

Fixes https://github.com/coala/projects/issues/96

To open a project's detail view, we've to write full name of project in lower case with underscore in place of space.
http://projects.coala.io#improve_diff_handling 

To search something in search box
http://projects.coala.io#search=improve

### CODE REVIEW

File: resources/js/app.js [hunk 1]
```diff
@@ -58,6 +98,14 @@
         };
     });
 
+	app.filter('format_name', function () {
+        return function (value) {
+        	  if(value){
```
Comment (sils) [inline resources/js/app.js]: please indent properly

File: resources/js/app.js [hunk 2]
```diff
@@ -11,20 +11,60 @@
...
 					});
 					$scope.currentProject = project
 				}
 
+				/* If the url has parameters, then this section is
+				executed. It checks whether url has a search query
+				or is it project url. */ 
+				var searchObject = $location.url();
+				if(searchObject){
+				searchObject = searchObject.split('=');
```
Comment (sils) [inline resources/js/app.js]: indentation...

**Comment by sils** [comment]
the code is super hard to read, doesn't coala have something to autoformat it properly with like 2 spaces indentation or 4? Indentation is still off in a lot of places and it uses tabs which means 8 on github...

File: resources/js/app.js [hunk 3]
```diff
@@ -11,24 +11,64 @@
...
-			controllerAs: 'lc'
-		}
-	}]);
+				/* If the url has parameters, then this section is
+				executed. It checks whether url has a search query
+				or is it project url. */ 
+				var searchObject = $location.url();
+				if(searchObject){
+					searchObject = searchObject.split('=');
+					if(searchObject.length == 2 && searchObject[0].indexOf('search') ==1){
```
Comment (sils) [inline resources/js/app.js]: there must be a smarter way to get the URL parsed, it should be easy to do it with `#search?q=searchterm` then angular knows you want to the `search` page and parses the GET parameters

Comment (hemangsk) [inline resources/js/app.js]: Yes That'd be much better
project.coala.io#?search=improve
and for project too?
project.coala.io#?project=improve_diff_handling

Comment (sils) [inline resources/js/app.js]: you'll wanna use angular routes for the pages, e.g. `#projects` for the projects view `#mentors` for the mentors page. Then use the get parameters for any parameters on the same page, `?q=...` or `?project=...`

Comment (hemangsk) [inline resources/js/app.js]: Figured these ones
```
project.coala.io#?search=improve
project.coala.io#?project=improve_diff_handling
```
will probably don't need a major redesign.

But having #project, #mentors is requiring major redesign in app.js file, shall we go with routing ?
 These ones  ```project.coala.io#?search=improve```  are also relatively shorter 😉

Comment (sils) [inline resources/js/app.js]: if it's a lot of effort concentrate on other things, this is only medium prio, other stuff would be more important

**Comment by hemangsk** [comment]
Closing because this approach doesn't fit right

**Pull request closed without merging**